remoting/src/main/java/org/apache/rocketmq/remoting/protocol/heartbeat/SubscriptionData.java

```diff
@@ -38,6 +38,9 @@ public class SubscriptionData implements Comparable<SubscriptionData> {
     @JSONField(serialize = false)
     private String filterClassSource;
 
+    @JSONField(serialize = false)
+    private transient long lastUpdateTime = System.currentTimeMillis();
```

Comment:
Should we use nanoTime?

Reply:
I don’t think it’s necessary. The lastUpdateTime is generated on the server where the Broker is located. Even if the client generates it, it will not be serialized and sent to the Broker, so there is no problem with changes